# HG changeset patch
# User Cole Robinson <crobinso@redhat.com>
# Date 1273684897 14400
# Node ID 45ba3080783a8b9e8b7a3796cc952e39b2093004
# Parent  affa97c8ae9a5c6e50d54b27997fac5f013916ea
engine: Add some more kvm default URI checks

diff -r affa97c8ae9a -r 45ba3080783a src/virtManager/engine.py
--- a/src/virtManager/engine.py	Wed May 12 12:57:32 2010 -0400
+++ b/src/virtManager/engine.py	Wed May 12 13:21:37 2010 -0400
@@ -66,9 +66,11 @@
     tryuri = None
     if os.path.exists("/var/lib/xend") and os.path.exists("/proc/xen"):
         tryuri = "xen:///"
-    elif (os.path.exists("/usr/bin/qemu") or
+    elif (os.path.exists("/dev/kvm") or
+          os.path.exists("/usr/bin/qemu") or
           os.path.exists("/usr/bin/qemu-kvm") or
-          os.path.exists("/usr/bin/kvm")):
+          os.path.exists("/usr/bin/kvm") or
+          os.path.exists("/usr/libexec/qemu-kvm")):
         tryuri = "qemu:///system"
 
     return tryuri
# HG changeset patch
# User Cole Robinson <crobinso@redhat.com>
# Date 1273686106 14400
# Node ID 18cffe6151ffa6e3e578ef4fb02e8fb00e93abd0
# Parent  45ba3080783a8b9e8b7a3796cc952e39b2093004
engine: Allow using older PackageKit search API

diff -r 45ba3080783a -r 18cffe6151ff src/virtManager/engine.py
--- a/src/virtManager/engine.py	Wed May 12 13:21:37 2010 -0400
+++ b/src/virtManager/engine.py	Wed May 12 13:41:46 2010 -0400
@@ -85,6 +85,7 @@
     Returns (success, did we just install libvirt) otherwise.
     """
     if not PACKAGEKIT_PACKAGES:
+        logging.debug("No PackageKit packages to search for.")
         return
 
     logging.debug("Asking PackageKit what's installed locally.")
@@ -191,7 +192,14 @@
     pk_trans.connect_to_signal('Finished', finished)
     pk_trans.connect_to_signal('ErrorCode', error)
     pk_trans.connect_to_signal('Package', package)
-    pk_trans.SearchNames("installed", [package_name])
+    try:
+        pk_trans.SearchNames("installed", [package_name])
+    except dbus.exceptions.DBusException, e:
+        if e.get_dbus_name() != "org.freedesktop.DBus.Error.UnknownMethod":
+            raise
+
+        # Try older search API
+        pk_trans.SearchName("installed", package_name)
 
     # Call main() so this function is synchronous
     gtk.main()
